pstress/generate_mysqld_options.sh

```diff
@@ -0,0 +1,511 @@
+#!/bin/bash
+# Created by Roel Van de Paar, Percona LLC
+
+# This script quickly and intelligently generates all available mysqld --option combinations (i.e. including values)
+
+# User variables
+OUTPUT_FILE=/tmp/mysqld_options_ms_80.txt
+
+# Internal variables, do not change
+TEMP_FILE=/tmp/mysqld_options.tmp
+
+if [ ! -r ./bin/mysqld ]; then
+  if [ ! -r ./mysqld ]; then
+    echo "This script quickly and intelligently generates all available mysqld --option combinations (i.e. including values)"
+    echo "Error: no ./bin/mysqld or ./mysqld found!"
+    exit 1
+  else
+    cd ..
+  fi
+fi
+
+IS_PXC=0
+if ./bin/mysqld --version | grep -q 'Percona XtraDB Cluster' 2>/dev/null ; then 
+  IS_PXC=1
+fi
+
+echoit(){
+  echo "[$(date +'%T')] $1"
+  echo "[$(date +'%T')] $1" >> /tmp/generate_mysqld_options.log
+}
+
+# Extract all options, their default values, and do some initial cleaning
+./bin/mysqld --no-defaults --help --verbose 2>/dev/null | grep -v "slave" 2>/dev/null | sed '0,/Value (after reading options)/d' |  egrep -v "To see what values.*is using|mysqladmin.*instead of|^[ \t]*$|\-\-\-" > ${TEMP_FILE}
```

Comment:
Why are you adding dev/null twice ?

Reply:
fixed

Reply:
In principle, each command which has the potential to, or commonly does, write to stderr, even if piped, should be redirected. If I put it there originally, and I did, then there was likely a reason for it.

Reply:
In principle, each command which has the potential to, or commonly does, write to stderr, even if piped, should be redirected. If I put it there originally, and I did for mysqld, then there was likely a reason for it.

Reply:
Hi Roel. Nice to see you actively looking at the changes and giving your valuable feedback. The orginal content looked like this:

`./bin/mysqld --no-defaults --help --verbose 2>/dev/null  | sed '0,/Value (after reading options)/d' | egrep -v "To see what values.*is using|mysqladmin.*instead of|^[ \t]*$|\-\-\-"  > ${TEMP_FILE}`

We have just added `grep -v "slave"`

`./bin/mysqld --no-defaults --help --verbose | grep -v "slave" 2>/dev/null  | sed '0,/Value (after reading options)/d' | egrep -v "To see what values.*is using|mysqladmin.*instead of|^[ \t]*$|\-\-\-"  > ${TEMP_FILE}`

Reply:
Hi Mohjit,

No worries. 

No, rather, you removed `2>/dev/null` from the mysqld call. Again, stderr is applicable to each command in a pipe. Adding it to the grep is a different thing altogether.

Consider the differences in output of stderr and stdout between:
```
$ ls notthere 2>/dev/null
$ ls notthere | grep -v 'cannot' 2>/dev/null
$ ls notthere 2>/dev/null | grep -v 'cannot'
$ ls notthere 2>/dev/null | grep -v 'cannot' 2>/dev/null
```
And you can understand. Each command here is distinctly different in how it operates.